varats/varats/ts_utils/click_param_types.py

```diff
@@ -174,11 +175,11 @@ def create_multi_experiment_type_choice(
     """
     initialize_experiments()
     value_dict = {
-        k: [v]
+        k: [v]  # make value a list to be consistent with entry for 'all'
         for k, v in ExperimentRegistry.experiments.items()
         if not __is_experiment_excluded(k)
     }
-    value_dict["all"] = list(value_dict.values())
+    value_dict["all"] = [v[0] for v in value_dict.values()]
```

Comment:
Better properly flatten the list:
```suggestion
    value_dict["all"] = [experiment for value in value_dict.values() for experiment in value]
```

Reply:
In my opinion, the two intermingled list comprehensions are quite hard to understand and somewhat counter-intuitive. As the definition of value_dict is just above and therefore not defined somewhere outside this function, I prefer my simpler expression. If you want to have this rewritten, I would propose to do it as a for loop explicitly concatenating arrays via the `+=` operator instead.

Reply:
Nested list comprehension are quite common in python and list comprehensions are also deemed to be "pythonic". After you know how they are build, they are often super easy to read/understand.
We also use them all over the place in the TS, so I would prefer to keep it.

Reply:
I don't like loops for such simple tasks, they take up so much space :smiley: 
But that's just my opinion.
I still vote for fully flattening the list since that respects the semantics of the type rather than some implementation detail on our side.